plugin/pkg/global/deletionconfirmation/admission.go

```diff
@@ -168,6 +197,26 @@ func (d *DeletionConfirmation) Validate(ctx context.Context, a admission.Attribu
 		}
 		checkFunc = gutil.CheckIfDeletionIsConfirmed
 
+	case v1alpha1.Kind("ShootState"):
+		listFunc = func() ([]client.Object, error) {
+			list, err := d.shootStateLister.List(labels.Everything())
```

Comment:
I just realized, having a watch for all `ShootState`s is probably not a good idea, as it will increase the api server's memory footprint even more (`ShootState`s are large resources).
We should probably use a metadata-only watch/client here, which is sufficient for this admission plugin.

Reply:
@timebertt I assume the API server already watches `ShootState`s anyways, doesn't it? How does it respond to clients reading `ShootState`s?

Reply:
Hmm, yes indeed. In the old `ShootState` plugin, g-api is already watching `ShootState`s: https://github.com/gardener/gardener/blob/2fb3b07c5c741b432889ce155a89a82cd88532b1/plugin/pkg/shootstate/validator/admission.go#L81

Still, this would be a very good opportunity for improvement.

> How does it respond to clients reading `ShootState`s?

The API server doesn't watch (via the k8s API) all resources by default. It only opens watches to etcd, if some client opens an API watch. But as long as clients only read from the API, there is no watch connection, IIRC.

Reply:
@timebertt and me agreed upon implementing a watch on the metadata of the ShootState. Due to their size ShootStates watches impose a signifacant amount of memory usage on APIservers in landscapes with a lot of Shoots.

Reply:
@BeckerMax and I concluded the following: although we see a significant benefit in using metadata-only watches or not watching shootstates at all, we don't want to do it in this PR, as shootstates were already watched before, so this PR doesn't change anything wrt memory consumption.
We tried the following approaches:
- using the c-r cache for starting metadata-only watches: problem here is, that we can't use the loopback rest config in g-api, but with the kube-apiserver rest config we have a cyclic dependency on startup: g-api is not discoverable until the startup has finished, but startup only finishes once the `core.gardener.cloud` API group is discoverable in order to construct a metada-only watch for shootstates
- completely leave out cache lookups and only rely on live lookups: this means we would need to list all shootstates for `DELETECOLLECTION` calls, which will again be a huge performance drawback

The last option was to use `client-go` style metadata-only watches/clients, but we don't have any reference usage for it in g/g yet, so we didn't try it out, but it's probably the way to go here.

Long story short, we should definitely improve the memory footprint of g-api by not watching shootstates directly, but this is not quite straight-forward so we want to do it in another PR.

Reply:
Opened https://github.com/gardener/gardener/issues/4106 regarding this improvement. Resolving this.